File: base.net/src/main/java/alvin/net/MainActivity.java
package alvin.net;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;

import alvin.net.status.StatusMainActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);
    }

    @OnClick({
            R.id.btn_network_status,
            R.id.btn_http,
            R.id.btn_socket
    })
    public void onButtonsClicked(Button b) {
        Intent intent = null;
        switch (b.getId()) {
        case R.id.btn_network_status:
            intent = new Intent(this, StatusMainActivity.class);
            break;
        }
        if (intent != null) {
            startActivity(intent);
        }
    }
}

File: base.kotlin/src/main/java/alvin/base/kotlin/main/views/MainActivity.kt
package alvin.base.kotlin.main.views

import alvin.base.kotlin.R
import alvin.base.kotlin.butterknife.views.ButterKnifeActivity
import alvin.base.kotlin.dagger.views.DaggerActivity
import alvin.base.kotlin.dbflow.views.DBFlowActivity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.intentFor

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        arrayOf(btn_butter_knife, btn_dbflow, btn_dagger).forEach { it ->
            it.setOnClickListener {
                val intent: Intent? = when (it) {
                    btn_butter_knife -> intentFor<ButterKnifeActivity>()
                    btn_dbflow -> intentFor<DBFlowActivity>()
                    btn_dagger -> intentFor<DaggerActivity>()
                    else -> null
                }
                if (intent != null) {
                    startActivity(intent)
                }
            }
        }
    }
}
